scalagen/shared/src/test/scala/org/scalameta/scalagen/Examples.scala

```diff
@@ -0,0 +1,36 @@
+package org.scalameta.scalagen
+
+import scala.meta._
+import scala.meta.contrib._
+import org.scalameta.scalagen.generators._
+
+/**
+  * Will print the structure of the class as it's toString.
+  * Not super useful, but a simple example to test with
+  */
+class StructuralToString extends ExtensionGenerator("StructuralToString") {
```

Comment:
I think having generators be classes that contain a 0-arg is a great convention. It makes it easy to classload. In scalafix I made the mistake of having SemanticdbIndex be passed in as a constructor argument, I wish I had made it a method `def withIndex(Index): Rule`

Reply:
I still have not figured out how params are going to work yet. It might be tricky